keydata.go

```diff
@@ -175,22 +273,35 @@ func (d *keyData) Marshal(w io.Writer) (nbytes int, err error) {
 	}
 
 	switch d.version {
-	case 0, 1:
+	case 0:
 		raw := keyDataRaw_v0{
 			KeyPrivate:        d.keyPrivate,
 			KeyPublic:         d.keyPublic,
 			AuthModeHint:      d.authModeHint,
 			StaticPolicyData:  makeStaticPolicyDataRaw_v0(d.staticPolicyData),
 			DynamicPolicyData: makeDynamicPolicyDataRaw_v0(d.dynamicPolicyData)}
 		n, err := tpm2.MarshalToWriter(w, raw)
-		nbytes += n
+		return nbytes + n, err
+	case 1:
+		var tmpW bytes.Buffer
+		raw := keyDataRaw_v1{
+			KeyPrivate:        d.keyPrivate,
+			KeyPublic:         d.keyPublic,
+			AuthModeHint:      d.authModeHint,
+			StaticPolicyData:  makeStaticPolicyDataRaw_v1(d.staticPolicyData),
+			DynamicPolicyData: makeDynamicPolicyDataRaw_v0(d.dynamicPolicyData)}
+		if _, err := tpm2.MarshalToWriter(&tmpW, raw); err != nil {
+			return 0, xerrors.Errorf("cannot marshal data: %w", err)
+		}
+		splitData, err := makeAfSplitData(tmpW.Bytes(), 128*1024, tpm2.HashAlgorithmSHA256)
 		if err != nil {
-			return nbytes, xerrors.Errorf("cannot marshal raw data: %w", err)
+			return 0, xerrors.Errorf("cannot split data: %w", err)
```

Comment:
Should these error cases `return` `0` or `nbytes`? The `version` code in `keydata.go` appeared to `return nbytes, ...` on error.

Reply:
Yeah, that should probably be nbytes, but this change in go-tpm2 will make this unnecessary:

https://github.com/canonical/go-tpm2/commit/bdbb15b0f6937b4f68708bb533482c144c5071e2

Reply:
Oh nice, this feels like a really good cleanup, it removes something that has confused me more than once! Thanks